src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -224,48 +224,63 @@ private static AnalyzerLoadFailureEventArgs CreateAnalyzerFailedArgs(Exception e
         {
             using var assembly = AssemblyMetadata.CreateFromFile(fullPath);
 
-            // This is longer than strictly necessary to avoid thrashing the GC with string allocations
-            // in the call to GetFullyQualifiedTypeNames. Specifically, this checks for the presence of
-            // supported languages prior to creating the type names.
-            var typeNameMap = from module in assembly.GetModules()
-                              from typeDefHandle in module.MetadataReader.TypeDefinitions
-                              let typeDef = module.MetadataReader.GetTypeDefinition(typeDefHandle)
-                              let supportedLanguages = GetSupportedLanguages(typeDef, module.Module, attributeType, languagesFunc)
-                              where supportedLanguages.Any()
-                              let typeName = GetFullyQualifiedTypeName(typeDef, module.Module)
-                              from supportedLanguage in supportedLanguages
-                              group typeName by supportedLanguage;
-
-            return typeNameMap.ToImmutableSortedDictionary(g => g.Key, g => g.ToImmutableHashSet(), StringComparer.OrdinalIgnoreCase);
+            Dictionary<string, ImmutableHashSet<string>.Builder> typeNameMap = new Dictionary<string, ImmutableHashSet<string>.Builder>(StringComparer.OrdinalIgnoreCase);
```

Comment:
Should we use `ImmutableSegmentedHashSet` instead?

Reply:
I didn't see these hitting the LOH in the profile that I looked at, so I'd prefer to keep as-is, without that evidence to back up such a change.